### PR TITLE
CDAP-4143 handle semicolons at the end of a query.

### PR DESCRIPTION
**Pull request by albertshau**
We have noticed that many users put a semicolon at the end of the
query. The db input formats don't work in such cases, which is
not very friendly. So check for semicolon and strip it if it exists.

### CODE REVIEW

File: database-plugins/src/test/java/co/cask/hydrator/plugin/DBConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * the License.
+ */
+
+package co.cask.hydrator.plugin;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ */
```
Comment (bdmogal) [inline database-plugins/src/test/java/co/cask/hydrator/plugin/DBConfigTest.java:24]: nit: empty javadoc

File: database-plugins/src/main/java/co/cask/hydrator/plugin/DBConfig.java [hunk 2]
```diff
@@ -62,4 +62,9 @@
   public DBConfig() {
     jdbcPluginType = "jdbc";
   }
+
+  protected String cleanQuery(String query) {
+    query = query.trim();
+    return query.endsWith(";") ? query.substring(0, query.length() - 1) : query;
```
Comment (bdmogal) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBConfig.java]: do you think we should handle multiple semi-colons? Or is that overkill?

Also, is the problem with only semicolons or with a set of characters at the end?

**Comment by bdmogal** [comment]
couple questions, other than that LGTM

File: database-plugins/src/main/java/co/cask/hydrator/plugin/DBConfig.java (hunk 2 above)
Comment (albertshau) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBConfig.java]: eh might as well handle multiple semi-colons.  Not sure what you mean by set of characters though?

Comment (bdmogal) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBConfig.java]: I meant do we need to handle any other special characters?

Comment (albertshau) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBConfig.java]: i don't know of anything else

Comment (bdmogal) [inline database-plugins/src/main/java/co/cask/hydrator/plugin/DBConfig.java]: :+1: can do it if we see a use case

**Comment by bdmogal** [comment]
:+1:

**Pull request merged**